src/components/Identity/createLegacyIdentity.js

```diff
@@ -48,21 +51,21 @@ export default ({ idMigrationEnabled, orgId, logger }) => {
   };
 
   return {
-    getEcidFromLegacy() {
+    getEcid() {
       if (idMigrationEnabled) {
         const ecid = getEcidFromLegacyCookies();
 
         if (ecid) {
           return Promise.resolve(ecid);
         }
 
-        if (doesVisitorExist) {
-          return getVisitorECID({ logger, orgId, awaitVisitorOptIn });
+        if (Visitor) {
```

Comment:
The only thing you are using Visitor for is to check if it exists.  If you change getEcidFromVisitor to return Promise.resolve() if Visitor is not there, then you don't need to have getVisitor as a dependency of createLegacyIdentity.js